Source/WebCore/platform/graphics/nicosia/NicosiaBuffer.cpp

```diff
@@ -119,6 +112,20 @@ UnacceleratedBuffer::~UnacceleratedBuffer()
     }
 }
 
+#if USE(SKIA)
+SkCanvas* UnacceleratedBuffer::canvas()
+{
+    if (m_surface)
+        return m_surface->getCanvas();
+
+    auto imageInfo = SkImageInfo::Make(m_size.width(), m_size.height(), kRGBA_8888_SkColorType, kPremul_SkAlphaType, SkColorSpace::MakeSRGB());
+    // FIXME: ref buffer and unref on release proc?
+    SkSurfaceProps properties = { 0, FontRenderOptions::singleton().subpixelOrder() };
+    m_surface = SkSurfaces::WrapPixels(imageInfo, m_data.get(), imageInfo.minRowBytes64(), &properties);
+    return m_surface ? m_surface->getCanvas() : nullptr;
```

Comment:
Why do we want to keep the surface alive if it doesn't have a canvas?

Reply:
Ah, I see, to avoid recreating the surface if that's failing to create a canvas.